FreeRTOS-Plus/Demo/AWS/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/ShadowDemoMainExample.c

```diff
@@ -607,205 +796,291 @@ void prvShadowDemoTask( void * pvParameters )
-                                        SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( democonfigTHING_NAME ),
-                                        SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
-    }
+        if( xDemoStatus == pdPASS )
+        {
+            /* Try to subscribe to `/delete/rejected` topic. */
+            xDemoStatus = xSubscribeToTopic( &xMqttContext,
+                                             SHADOW_TOPIC_STRING_DELETE_REJECTED( democonfigTHING_NAME ),
+                                             SHADOW_TOPIC_LENGTH_DELETE_REJECTED( THING_NAME_LENGTH ) );
+        }
 
-    if( demoStatus == pdPASS )
-    {
-        demoStatus = xSubscribeToTopic( &xMqttContext,
-                                        SHADOW_TOPIC_STRING_UPDATE_REJECTED( democonfigTHING_NAME ),
-                                        SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
-    }
+        if( xDemoStatus == pdPASS )
+        {
+            /* Publish to Shadow `delete` topic to attempt to delete the
+             * Shadow document if exists. */
+            xDemoStatus = xPublishToTopic( &xMqttContext,
+                                           SHADOW_TOPIC_STRING_DELETE( democonfigTHING_NAME ),
+                                           SHADOW_TOPIC_LENGTH_DELETE( THING_NAME_LENGTH ),
+                                           pcUpdateDocument,
+                                           0U );
+        }
 
-    /********************* Publish to Shadow topics. **********************/
-
-    /* This demo uses a constant #democonfigTHING_NAME known at compile time
-     * therefore we can use macros to assemble shadow topic strings.
-     * If the thing name is known at run time, then we could use the API
-     * #Shadow_GetTopicString to assemble shadow topic strings, here is the
-     * example for /update/delta:
-     *
-     * For /update/delta:
-     *
-     * #define SHADOW_TOPIC_MAX_LENGTH  (256U)
-     *
-     * ShadowStatus_t shadowStatus = SHADOW_STATUS_SUCCESS;
-     * char cTopicBuffer[ SHADOW_TOPIC_MAX_LENGTH ] = { 0 };
-     * uint16_t usBufferSize = SHADOW_TOPIC_MAX_LENGTH;
-     * uint16_t usOutLength = 0;
-     * const char * pcThingName = "TestThingName";
-     * uint16_t usThingNameLength  = ( sizeof( pcThingName ) - 1U );
-     *
-     * shadowStatus = Shadow_GetTopicString( SHADOW_TOPIC_STRING_TYPE_UPDATE_DELTA,
-     *                                       pcThingName,
-     *                                       usThingNameLength,
-     *                                       & ( cTopicBuffer[ 0 ] ),
-     *                                       usBufferSize,
-     *                                       & usOutLength );
-     */
+        /* Wait for an incoming publish on `/delete/accepted` or `/delete/rejected`
+         * topics, if not already received a publish. */
+        if( ( xDemoStatus == pdPASS ) && ( xDeleteResponseReceived != pdTRUE ) )
```

Comment:
Can the second expression ever be false?

Reply:
Yes. The function xPublishToTopic has a call to MQTT_ProcessLoop() which may result in receiving an incoming publish on any of the `/delete/accepted` or `/delete/rejected` topics and will set the `xDeleteResponseReceived` to `pdTRUE`.